api_tests/logs/views/test_log_detail.py

```diff
@@ -1,142 +1,200 @@
 
-        self.node_log_url = '/{}nodes/{}/logs/'.format(API_BASE, self.node._id)
-        self.url = '/{}logs/'.format(API_BASE)
-        self.log_nodes_url = self.url + '{}/nodes/'.format(self.log._id)
-        self.private_log_detail = self.url + '{}/'.format(self.log._id)
-        self.log_public_nodes_url = self.url + '{}/nodes/'.format(self.public_log._id)
-        self.public_log_detail = self.url + '{}/'.format(self.public_log._id)
+    @pytest.fixture()
+    def log_public(self, logs_public):
+        return logs_public[0]
 
+    @pytest.fixture()
+    def contributor_log_public(self, logs_public):
+        return logs_public[1]
 
-class TestLogDetail(LogsTestCase):
-
-    def test_log_detail_returns_data(self):
-        res = self.app.get(self.private_log_detail, auth=self.user.auth)
-        assert_equal(res.status_code, 200)
-        json_data = res.json['data']
-        assert_equal(json_data['id'], self.log._id)
-
-    def test_log_detail_private_not_logged_in_cannot_access_logs(self):
-        res = self.app.get(self.private_log_detail, expect_errors=True)
-        assert_equal(res.status_code, 401)
+    @pytest.fixture()
+    def logs_private(self, node_private):
+        return list(node_private.logs.order_by('date'))
 
-    def test_log_detail_private_non_contributor_cannot_access_logs(self):
-        res = self.app.get(self.private_log_detail, auth=self.user_two.auth, expect_errors=True)
-        assert_equal(res.status_code, 403)
+    @pytest.fixture()
+    def log_private(self, logs_private):
+        return logs_private[0]
 
-    def test_log_detail_public_not_logged_in_can_access_logs(self):
-        res = self.app.get(self.public_log_detail, expect_errors=True)
-        assert_equal(res.status_code, 200)
-        json_data = res.json['data']
-        assert_equal(json_data['id'], self.public_log._id)
-
-    def test_log_detail_public_non_contributor_can_access_logs(self):
-        res = self.app.get(self.public_log_detail, auth=self.user_two.auth, expect_errors=True)
-        assert_equal(res.status_code, 200)
-        json_data = res.json['data']
-        assert_equal(json_data['id'], self.public_log._id)
+    @pytest.fixture()
+    def contributor_log_private(self, logs_private):
+        return logs_private[1]
 
-    def test_log_detail_data_format_api(self):
-        res = self.app.get(self.public_log_detail + '?format=api', auth=self.user.auth)
-        assert_equal(res.status_code, 200)
-        assert_in(self.public_log._id, unicode(res.body, 'utf-8'))
+    @pytest.fixture()
+    def url_node_private_log(self, node_private):
+        return '/{}nodes/{}/logs/'.format(API_BASE, node_private._id)
 
+    @pytest.fixture()
+    def url_logs(self):
+        return '/{}logs/'.format(API_BASE)
 
-class TestNodeFileLogDetail(ApiTestCase):
+    @pytest.fixture()
+    def url_log_private_nodes(self, log_private, url_logs):
+        return url_logs + '{}/nodes/'.format(log_private._id)
```

Comment:
format should probably be used instead of string concatenation
'{}{}/nodes/'.format(etc.)